### PR TITLE
feat : Add borrowing and returning functionality for books with avail…

### PR DESCRIPTION
**Pull request by olaiide**
…ability checks

### CODE REVIEW

File: controllers/bookController.js [hunk 1]
```diff
@@ -10,7 +11,7 @@ exports.addBook = catchAsync(async (req, res, next) => {
   }
   const newBook = await Book.create(req.body);
   res.status(201).json({
-    status: "Success",
+    status: constants.SUCCESS_TEXT,
```
Comment (zeedt) [inline controllers/bookController.js]: You don't need to say `SUCCESS_TEXT`. Instead just name the constant `SUCCESS`

File: controllers/bookController.js [hunk 2]
```diff
@@ -94,3 +95,47 @@ exports.deleteBook = catchAsync(async (req, res, next) => {
     data: null,
   });
 });
+exports.borrowBook = catchAsync(async (req, res, next) => {
+  const id = req.params.id;
+  const book = await Book.findById(id);
+  if (!book) {
+    return next(new AppError("No book found with that ID", 404));
+  }
+  if (book.available === false) {
```
Comment (zeedt) [inline controllers/bookController.js]: Why not do this?

```suggestion
  if (!book.available) {
```
That is a better representation

File: controllers/bookController.js [hunk 3]
```diff
@@ -94,3 +95,47 @@ exports.deleteBook = catchAsync(async (req, res, next) => {
...
+    },
+  });
+});
+exports.returnBook = catchAsync(async (req, res, next) => {
+  const id = req.params.id;
+  const book = await Book.findById(id);
+  if (!book) {
+    return next(new AppError("No book found with that ID", 404));
+  }
+  if (book.available === true) {
```
Comment (zeedt) [inline controllers/bookController.js]:
```suggestion
  if (book.available) {
```

File: controllers/userController.js [hunk 4]
```diff
@@ -77,3 +78,36 @@ exports.getAllUsers = catchAsync(async (req, res, next) => {
...
+  const decoded = await promisify(jwt.verify)(token, process.env.JWT_SECRET);
+
+  //check if user still exists
+  const currentUser = await User.findById(decoded.id);
+  if (!currentUser) {
+    return next(
+      new AppError("The user belonging to this token does no longer exist")
+    );
+  }
+  //check if user changed password after the token was issued
```
Comment (zeedt) [inline controllers/userController.js]: Delete commented code

**Review by zeedt — CHANGES_REQUESTED** [review]
Address the minor comments.

Also, how do you intend to handle concurrency? What happens if two people request for the same book? Do you know one will override the other and they will both get successful response?

You don't need to handle it in this PR but read on how to handle that and make it your next tak

**Comment by olaiide** [comment]
> Address the minor comments.
> 
> Also, how do you intend to handle concurrency? What happens if two people request for the same book? Do you know one will override the other and they will both get successful response?
> 
> You don't need to handle it in this PR but read on how to handle that and make it your next tak

alright, i will

**Pull request merged**